### DISCUSSION TITLE
Questions about widgets and styles

### DISCUSSION THREAD

**Discussion by CarlosHGL** [beeware/toga#1752] [Q&A] [answered]
question about toga-android, can I add a corner_radius in the widget editing or does toga not support that yet? And is there any scheduled date for the implementation of the Frame widget in toga-android? The frames would be very convenient for my current project along with the corner-radius.

**Comment by freakboy3742** [comment, marked as answer]
`corner-radius` isn't something that is supported anywhere in Toga - and, broadly speaking, it would be unlikely to *ever* be supported. Toga very strongly supports using the platform's look-and-feel; unless I'm misunderstanding, `corner-radius` is a setting that is part of the platform API providing flexibility to control the specific appearance of widgets.

As for Frame - I don't know specifically which widget you're referring to here, as I can't find any reference to an `android.widgets.Frame`. If you mean [FrameLayout](https://developer.android.com/reference/android/widget/FrameLayout), it's unlikely to ever be supported directly, as Toga has its own layout mechanism.

That said - one of the features of Toga is that you always have access to the underlying widgets; `widget._impl.native` will usually be a reference to the underlying native widget implementing an API. So, if you want to set a corner radius, you can, by writing directly to the widget (at the cost of your code no longer being platform independent). Using a FrameLayout may also be possible, but will be a lot more complicated due to the interaction with Toga's layout system.

If you have interest in specific Android APIs, you may want to consider using the Chaquopy layer directly, rather than using Toga as a wrapper. That won't let you build cross-platform apps, but if you're thinking about your app in terms of specific Android APIs, it may be a closer match.
